src/luarocks/cmd/install.lua

```diff
@@ -208,51 +212,46 @@ local function install_rock_file(filename, opts)
 end
 
 --- Driver function for the "install" command.
--- @param name string: name of a binary rock. If an URL or pathname
--- to a binary rock is given, fetches and installs it. If a rockspec or a
--- source rock is given, forwards the request to the "build" command.
+-- If an URL or pathname to a binary rock is given, fetches and installs it.
+-- If a rockspec or a source rock is given, forwards the request to the "build"
+-- command.
 -- If a package name is given, forwards the request to "search" and,
 -- if returned a result, installs the matching rock.
--- @param version string: When passing a package name, a version number
--- may also be given.
 -- @return boolean or (nil, string, exitcode): True if installation was
 -- successful, nil and an error message otherwise. exitcode is optionally returned.
-function install.command(flags, name, version)
-   if type(name) ~= "string" then
-      return nil, "Argument missing. "..util.see_help("install")
-   end
-
-   name = util.adjust_name_and_namespace(name, flags)
+function install.command(args)
+   args.rock = util.adjust_name_and_namespace(args.rock, args)
```

Comment:
I'm wondering if we should replace this with something in the argument parsing. Or alternatively just have the function return `namespace, name`.

Reply:
`adjust_name_and_namespace` returns a name in the user/rock format. It might be better to keep the name and namespace separate to avoid splitting it again in e.g. `queries.new`. The argparse action callback could split the name and namespace, putting the name in `args.name` and the namespace in `args.namespace`.

Reply:
Sounds like a good idea to me. But lets do that in another PR after this one is merged?

Reply:
Tracked in #1067